### PR TITLE
Bump eventing components v1.11

### PR DESCRIPTION
**Pull request by pierDipi**
- Bump eventing components to 1.11
- Implement logic to reconcile TLS resources (cert-manager's certificates and issuers)

### CODE REVIEW

**Comment by pierDipi** [comment]
```
  * could not run steps: step [release:latest] failed: failed to get CLI image: unable to wait for the 'cli' image in the stable stream to populate (waited for 5m5.019479059s): timed out waiting for the condition 
INFO[2023-10-13T16:38:44Z] Reporting job state 'failed' with reason 'executing_graph:step_failed:importing_release' 
```
/retest

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.11/3-in-memory-channel.yaml [hunk 1]
```diff
@@ -203,7 +232,7 @@ spec:
               fieldRef:
                 fieldPath: metadata.namespace
           - name: DISPATCHER_IMAGE
-            value: registry.ci.openshift.org/openshift/knative-v1.10:knative-eventing-channel-dispatcher
+            value: registry.ci.openshift.org/openshift/knative-v1.11:knative-eventing-channel-dispatcher
```
Comment (pierDipi) [inline openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.11/3-in-memory-channel.yaml:235]: @matzew should this be a "TO BE REPLACED" as well? How does this image replacement work?

File: knative-operator/pkg/controller/knativekafka/eventing_tls.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"knative.dev/eventing/pkg/apis/feature"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	serverlessoperatorv1alpha1 "github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis/operator/v1alpha1"
+)
+
+var (
+	tlsResourcesPred = byGroup("cert-manager.io")
```
Comment (matzew) [inline knative-operator/pkg/controller/knativekafka/eventing_tls.go:17]: what is `pred` ?

File: test/eventing-kafka.bash [hunk 3]
```diff
@@ -30,8 +30,7 @@ function upstream_knative_eventing_kafka_broker_e2e {
 
   export TEST_IMAGE_TEMPLATE="registry.ci.openshift.org/openshift/knative-eventing-kafka-broker-test-{{.Name}}:${KNATIVE_EVENTING_KAFKA_BROKER_VERSION}"
 
-  GORUN_PATH="$(mktemp -t -d -u gopath.XXXXXXXX)"
-  export GORUN_PATH
+  export SKIP_GENERATE_RELEASE=true
```
[the remark this answers is not in the document]
Comment (pierDipi) [reply test/eventing-kafka.bash:33]: it's a flag to skip generate release

**Pull request merged**